include/nlohmann/json.hpp

```diff
@@ -3817,17 +3736,18 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
     with range checking
     @sa see @ref value() for access by value with a default value
 
-    @since version 1.1.0
+    @since version 1.0.0; template KeyT added in version 3.10.0
     */
-    template<typename T>
-    JSON_HEDLEY_NON_NULL(2)
-    const_reference operator[](T* key) const
+    template < class KeyT, typename detail::enable_if_t <
+                   detail::is_key_type<basic_json_t, KeyT>::value&& !std::is_same<typename std::decay<KeyT>::type, json_pointer>::value > ... >
+    const_reference operator[](KeyT&& key) const
```

Comment:
Since your semantics of the const version of operator[] is that the key *MUST* be found in the map,

you don't want && here. you want const&, since you don't want to construct a new item if the item isn't found.

That's different from the non-const version, of course, where you *do* want a new item created.